Gordon360/Controllers/RecIM/ActivitiesController.cs

```diff
@@ -0,0 +1,102 @@
+using Gordon360.Services.RecIM;
+using Gordon360.Authorization;
+using Gordon360.Static.Names;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    [AllowAnonymous]
+    public class ActivitiesController : GordonControllerBase
+    {
+        private readonly IActivityService _activityService;
+
+        public ActivitiesController(IActivityService activityService)
+        {
+            _activityService = activityService;
+        }
+
+        ///<summary>Gets a list of all Activities by parameter </summary>
+        ///<param name="active"> Optional active parameter </param>
+        ///<param name="time"> Optional time parameter </param>
+        /// <returns>
+        /// All Existing Activities 
+        /// </returns>
+        [HttpGet]
+        [Route("")]
+        public ActionResult<IEnumerable<ActivityExtendedViewModel>> GetActivities([FromQuery] DateTime? time, bool active)
+        {   
+            if (time is null && active)
+            {
+                var activeResults = _activityService.GetActivities();
+                return Ok(activeResults);
+
+            }
+            var result = _activityService.GetActivitiesByTime(time);
+            return Ok(result);
+        }
+
+        ///<summary>Gets a Activity object by ID number</summary>
+        /// <param name="activityID">League ID Number</param>
+        /// <returns>
+        /// Activity object
+        /// </returns>
+        [HttpGet]
+        [Route("{activityID}")]
+        public ActionResult<ActivityExtendedViewModel> GetActivityByID(int activityID)
+        {
+            var result = _activityService.GetActivityByID(activityID);
+            return Ok(result);
+        }
+
+        [HttpGet]
+        [Route("lookup")]
+        public ActionResult<IEnumerable<LookupViewModel>> GetActivityTypes(string type)
+        {
+            var res = _activityService.GetActivityLookup(type);
+            if (res is not null)
+            {
+                return Ok(res);
+            }
+            return BadRequest();
+        }
+
+        /// <summary>
+        /// Posts Activity into CCT.RecIM.Activity
+        /// </summary>
+        /// <param name="newActivity">CreateActivityViewModel object with appropriate values</param>
+        /// <returns>Posted Activity</returns>
+        [HttpPost]
+        [Route("")]
+        [StateYourBusiness(operation = Operation.ADD, resource = Resource.RECIM_ACTIVITY)]
+        public async Task<ActionResult<ActivityViewModel>> CreateActivity(ActivityUploadViewModel newActivity)
+        {
+            var activity = await _activityService.PostActivityAsync(newActivity);
+            return CreatedAtAction("CreateActivity", activity);
```

Comment:
This is not how `CreatedAtAction` is intended to be used. The first parameter is the location (i.e. URL) that the newly created object can be retrieved from. 

In this case, it should look like this:
```suggestion
            return CreatedAtAction(nameof(GetActivityByID), new { activityID = acitivty.ID}, activity);
```

Reply:
I see, I'll make this change globally